update ubuntu 22.04 to 24.04

ViniRodrig opened the pull request:
- charmcraft 3.0 base (deprecated) -> platform
-  update ops lib v0 apt
- tests charm update to noble
- vault update in the bundle

ViniRodrig committed:
chore: charmcraft 3.0 base (deprecated) -> platform; update ops lib v0 apt

feat: tests noble charm update

vault update

Signed-off-by: Vinicius Rodrigues <vmarquesrodrigues25@gmail.com>

luissimas committed:
test: assert that failure domain is other than the default on all members

When we first wrote these tests, the LXD cloud (which we use in our CI)
did not support availability zones. For this reason, we asserted that
all nodes had the "default" failure domain.

Juju 3.6.9 introduced support for availability zones in LXD clouds and
sets them to the host name that is running the LXD VMs. With that in
mind, we need to update the integration test to instead assert that we
have a failure domain different than the default. Since our CI creates
LXD nodes on-demand, it makes little sense to match an specific
hostname.

References:
https://documentation.ubuntu.com/juju/3.6/reference/juju/juju-roadmap-and-releases/#lxd

Signed-off-by: Luís Simas <luissimas@protonmail.com>

luissimas commented on tests/integration/bundles/incus.yaml.j2:
```diff
@@ -17,7 +17,7 @@ applications:
   vault:
     charm: vault
-    channel: 1.8/stable
+    channel: latest/edge
```
AFAIK we can't use the latest vault charm version because they completely changed the certificates relation interface, making it incompatible with the OVN charms. Could you explain why we needed to update this?

The latest version we can use is `1.8`. Unless the OVN charms have been updated to deal with the new interface. If that's the case, then we'd also need to update the incus charm itself to support this new certificates interface.

See more here:

https://canonical-vault-charms.readthedocs-hosted.com/en/1.17/reference/key_differences_between_charm_versions/

luissimas replied:
Nvm, it seems like `latest/edge` is the channel for the old vault charm, while the `1.15/*` channels are for the new charm. We should be fine with this channel.

luissimas commented on tests/integration/test_charm_clustered.py:
```diff
@@ -680,7 +680,7 @@ async def test_create_instance(ops_test: OpsTest):
     target_node = ops_test.model.units["incus/3"].machine.hostname
     action = await unit.run(
-        f"incus launch images:ubuntu/22.04 {TEST_INSTANCE_NAME} --vm --storage ceph --quiet --target {target_node}"
+        f"incus launch images:ubuntu/24.04 {TEST_INSTANCE_NAME} --vm --storage ceph --quiet --target {target_node}"
```
Now I'm trying to remember why we use Ubuntu images for those test instances that we create to verify that the cluster deployed is functional.

I think that maybe we could just use an Alpine or Debian image to save a few seconds, potentially minutes in the CI. WDYT?

But this is totally unrelated to this PR, so it's not really a block for merging this.

luissimas committed:
ci: move scheduled workflow from friday to monday

We're more likely to investigate issues during the week :)

Signed-off-by: Luís Simas <luissimas@protonmail.com>

luissimas committed:
ci: explicitly install required tools

Explicitly install all required CI dependencies on the runners. Before
this, we were using a custom image based on Ubuntu Jammy that had some
packages pre-installed. By making everything explicitly installed in the
workflow file, and depending only on a base Ubuntu Noble image, we
increase the portability of the CI for other environments.

Signed-off-by: Luís Simas <luissimas@protonmail.com>

luissimas committed:
docs: add ci docs

Signed-off-by: Luís Simas <luissimas@protonmail.com>

luissimas committed:
test: temporarily disable failure domain setting in both tests

Signed-off-by: Luís Simas <luissimas@protonmail.com>

luissimas committed:
test: use 20G root disk for new unit in clustered tests

We need to explicitly set this size in the new instance, otherwise the
tests will fail as there will be not enough space to create the Incus
storage pool in the new member.

Signed-off-by: Luís Simas <luissimas@protonmail.com>
